doc/dev/index.rst

```diff
@@ -178,12 +178,75 @@ with:
     # check that it's there
     ./ceph health
 
+Architecture
```

Comment:
I think this section does not belong here but in a different part of the documentation.